src/components/CurrentPlaylist.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { TrackItem } from './TrackItem';
+import { ErrorMessage } from './ErrorMessage';
+
+class CurrentPlaylist extends React.Component {
+  state = {
+    error: false,
+    errorMessage: '',
+    tracks: {
+      items: [],
+    },
+    nextTracksEndpoint: null,
+  }
+
+  componentDidMount() {
+    const { token, playlist } = this.props;
+    this.fetchCurrentPlaylistTracks(token, playlist.tracks.href);
+  }
+
+  componentDidUpdate() {
+    const { token } = this.props;
+    const { nextTracksEndpoint } = this.state;
+
+    if (nextTracksEndpoint) {
+      this.fetchCurrentPlaylistTracks(token, nextTracksEndpoint);
+    }
+  }
+
+
+  fetchCurrentPlaylistTracks = (token, endpoint) => {

```

Comment:
Try to refactor this using async/await, and pull the network logic out into it's own function so you can invoke it and get the value you want in a line, rather than managing all the shit in the component